webpack/JobWizard/JobWizard.js

```diff
@@ -18,10 +18,19 @@ export const JobWizard = () => {
 
   const setDefaults = useCallback(response => {
     const responseJob = response.data;
-    setAdvancedValues({
+    const advancedTemplateValues = {};
+    const advancedInputs = responseJob.advanced_template_inputs;
+    if (advancedInputs) {
+      advancedInputs.forEach(input => {
+        advancedTemplateValues[input.name] = input?.default || '';
+      });
+    }
+    setAdvancedValues(currentAdvancedValues => ({
```

Comment:
can you use the same debounced setState here and also for the fields?
currently, they re-render the parent state quite a lot:

```
advancedValues {"effectiveUserValue":"a","timeoutToKill":"","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"ad","timeoutToKill":"","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"adm","timeoutToKill":"","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"admi","timeoutToKill":"","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"admin","timeoutToKill":"","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"admin","timeoutToKill":"1","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"admin","timeoutToKill":"12","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"admin","timeoutToKill":"12","templateValues":{"pre_script":"","post_script":""}}
JobWizard.js:18 advancedValues {"effectiveUserValue":"admin","timeoutToKill":"12","templateValues":{"pre_script":"","post_script":""}}
```

Reply:
What debounced setstate do you mean?

Reply:
you can use `lodash` for that: https://lodash.com/docs/4.17.15#debounce

Reply:
I tried to use it in the use state but it looks like it prevents the user from typing too fast.
I used 150 ms and when I tried to type in `1234567890-` it only recorded `-`
https://github.com/theforeman/foreman_remote_execution/blob/7e88ad1e4eaa05ca3dfeb70861606525a3757047/webpack/JobWizard/JobWizard.js#L71

Reply:
we discussed it offline and decided to do more research on it as a performance improvement,
note: maybe `throttle` func is better here